### PR TITLE
Add flag for ignoring mismatching device types

### PR DESCRIPTION
**Pull request by svenschwermer**

### CODE REVIEW

**Comment by svenschwermer** [comment]
I can't find a repository.yml that is supposed to hold missing or forbidden tags.

**Comment by zvin** [comment]
@svenschwermer you need to add a `Change-Type: patch|minor|major` flag as the last line of your commit.

I'd suggest using `patch` for this one as it doesn't change anything in `preload.js`.

**Comment by zvin** [comment]
@svenschwermer you'll also need to rebase on master.

**Comment by zvin** [comment]
@svenschwermer I'll add the same flag to [resin-cli](https://github.com/resin-io/resin-cli/blob/master/lib/actions/preload.coffee) as /bin/resin-preload is deprecated.

**Comment by svenschwermer** [comment]
@zvin Why do I need to rebase on master?

**Comment by zvin** [comment]
@svenschwermer because it has changed since your commit.

**Comment by svenschwermer** [comment]
Should be fixed now.

File: bin/resin-preload [hunk 1]
```diff
@@ -155,7 +159,7 @@ streamToPromise(buildOutputStream)
     return preload.getDeviceTypeSlugAndPreloadedBuilds(docker, options)
   })
   .then((data) => {
-    if (data.slug !== options.application.device_type) {
+    if (options.dontCheckDeviceType == false && data.slug !== options.application.device_type) {
```
Comment (zvin) [inline bin/resin-preload]: One last thing: please use `===` instead of `==`.

**Comment by zvin** [comment]
Please squash the 2 commits.

**Comment by zvin** [comment]
@svenschwermer changes merged.
They are in `v3.1.2`.
Thank you for your contribution.

**Pull request merged**